src/acrn/acrn_driver.c

```diff
@@ -2605,6 +2650,9 @@ acrnStateInitialize(bool privileged,
                                        NULL, NULL) < 0)
         goto cleanup;
 
+    virDomainObjListForEach(acrn_driver->domains, acrnPersistentDomainInit,
+        acrn_driver);
```

Comment:
If the return value is < 0, we should go to cleanup.

Reply:
A side comment is we're doing something potentially dangerous here, which stemmed from the implementation of acrnAllocateVm(). We hold the domain's read lock recursively during this call, but we're actually changing its content (priv->hvUUID). However, pthread_rwlock_wrlock() does not define the behavior of recursive write locks or lock upgrades. This only happens during acrnStateInitialize() right now, so we're not in danger of a race condition. But it's something we should be aware of.

Reply:
Using virDomainObjListForEach without checking the return value seems a common practice throughout libvirt and I think this is why I just missed it. But you are perfectly right, I'll add that check.